analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -175,6 +151,16 @@ internal static bool IsAnalysisScopeMatching(Compilation compilation, bool isTes
             return diagnostics.Any(d => d.CustomTags.Contains(matchingScopeTag));
         }
 
+        private static bool IsTestProject(TryGetValueDelegate<ProjectConfigReader> tryGetValue, Compilation compilation, AnalyzerOptions options)
+        {
+            var projectType = ProjectConfiguration(tryGetValue, options).ProjectType;
+            return projectType == ProjectType.Unknown
+                // SonarLint, NuGet or Scanner <= 5.0. We run the rule if the compilation is null, becase we can't detect test references.
+                ? compilation?.IsTest() ?? true
```

Comment:
I have a doubt about this `true` fallback. It's answering `IsTestProject` question so it should be true.
I came across this during preparation for the more UTs. As the comment about running the rule is also not relevant to this case (seems like an artifact from the `IsAnalysisScopeMatching` comment)

Reply:
Indeed it's from `IsAnalysisScopeMatching`

It shouldn't be `false` or `true`, is should just match the analysis scope. But it's better to be `false`, indeed.

Reply:
Update is on it's way in #4185